PDSim/core/StateFlooded.pyx

```diff
@@ -0,0 +1,786 @@
+        enthalpy and saturation temperature
+        Caveat: Function only good for pure refrigerant
+        """
+        """
+        When fsolve() is called, it will use a single element
+        ndarray but we only want a double to be passed along,
+        so take the first element of the 1-element array, which
+        is a 64-bit float
+        """
+        
+        print 'h',h
+        HEOS = CoolProp.AbstractState('HEOS',Ref)
+        HEOS.update.(CoolProp.TQ_INPUTS,T,0)
+        self.h_l = HEOS.keyed_output(CoolProp.iHmass) 
+        
+        HEOS = CoolProp.AbstractState('HEOS',Ref)
+        HEOS.update.(CoolProp.TQ_INPUTS,T,1)
+        self.h_v = HEOS.keyed_output(CoolProp.iHmass) 
+        
+        self.Qth = (h - self.h_l)/(self.h_v - self.h_l)
+        return self.Qth
+    
+    def Q_Ts(self,Ref,Liq,T,s,Q_guess):
+        
+        """
+        Solve for the two-phase quality of refrigerant in the dome for a given
+        enthalpy and saturation temperature
+        Caveat: Function only good for pure refrigerant
+        """
+        """
+        When fsolve() is called, it will use a single element
+        ndarray but we only want a double to be passed along,
+        so take the first element of the 1-element array, which
+        is a 64-bit float
+        """
+        HEOS = CoolProp.AbstractState('HEOS',Ref)
+        HEOS.update.(CoolProp.TQ_INPUTS,T,Q[0])
+        
+        self.f = lambda Q : HEOS.keyed_output(CoolProp.iSmass) - s
+        self.Q = optimize.fsolve(self.f,Q_guess)
+        return float(self.Q)
+    
+    
+    def T_crit(self,Ref):    # Critical Temperature of the refrigerant [K] - From coolprop  ->  Props(Fluid,PropName)
+        
+        if Ref=='R245fa':
+            HEOS = CoolProp.AbstractState('HEOS',Ref)
+            self.T = HEOS.keyed_output(CoolProp.Tcrit) 
+        elif Ref=='R410A':
+            HEOS = CoolProp.AbstractState('HEOS',Ref)
+            self.T = HEOS.keyed_output(CoolProp.Tcrit)
+        elif Ref=='R744':
+            self.T = 304.128
+        elif Ref=='R404A':
+            HEOS = CoolProp.AbstractState('HEOS',Ref)
+            self.T = HEOS.keyed_output(CoolProp.Tcrit)
+        elif Ref=='R134a':
+            HEOS = CoolProp.AbstractState('HEOS',Ref)
+            self.T = HEOS.keyed_output(CoolProp.Tcrit)      #Tcr=374.2
+        elif Ref=='SES36':
+            HEOS = CoolProp.AbstractState('HEOS',Ref)
+            self.T = HEOS.keyed_output(CoolProp.Tcrit)
+        elif Ref=='R290':
+            self.T = 369.82
+        elif Ref=='R717':
+            self.T = 405.4
+        else:
+            print 'Uh oh...  Refrigerant not found T_crit'
+        return self.T
+    
+    def p_crit(self,Ref):    # Critical pressure of the refrigerant [kPa] - From coolprop  ->  Props(Fluid,PropName)
+        if Ref=='R245fa':
+            HEOS = CoolProp.AbstractState('HEOS',Ref)
+            self.p = HEOS.keyed_output(CoolProp.pcrit)
+        elif Ref=='R410A':
```

Comment:
Not sure if it is necessary to define p_crit and T_crit within StateFlooded class, since it can be called from PDSim directly from AbstractState class.